indexer/src/main/java/org/uniprot/store/indexer/common/config/UniProtSolrClient.java

```diff
@@ -0,0 +1,160 @@
+        }
+    }
+
+    public <T> Optional<T> queryForObject(
+            SolrCollection collection, SolrQuery solrQuery, Class<T> returnType) {
+        String collectionString = collection.name();
+        try {
+            QueryResponse response =
+                    threadLocalSolrClients.get().query(collectionString, solrQuery);
+            List<T> beans = response.getBeans(returnType);
+            if (beans.size() == 1) {
+                return Optional.of(beans.get(0));
+            } else {
+                return Optional.empty();
+            }
+        } catch (SolrServerException | IOException e) {
+            throw new SolrQueryRetrievalException(
+                    "Could not write documents to Solr collection [" + collectionString + "]", e);
+        }
+    }
+
+    public void commit(SolrCollection collection) {
+        String collectionString = collection.name();
+        try {
+            threadLocalSolrClients.get().commit(collectionString);
+        } catch (SolrServerException | IOException e) {
+            throw new SolrQueryRetrievalException(
+                    "Could not commit contents of Solr collection [" + collectionString + "]", e);
+        }
+    }
+
+    public void softCommit(SolrCollection collection) {
+        String collectionString = collection.name();
+        try {
+            threadLocalSolrClients.get().commit(collectionString, true, true, true);
+        } catch (SolrServerException | IOException e) {
+            throw new SolrQueryRetrievalException(
+                    "Could not soft commit contents of Solr collection [" + collectionString + "]",
+                    e);
+        }
+    }
+
+    public void delete(SolrCollection collection, String query) {
+        String collectionString = collection.name();
+        try {
+            threadLocalSolrClients.get().deleteByQuery(collectionString, query);
+        } catch (SolrServerException | IOException e) {
+            throw new SolrQueryRetrievalException(
+                    "Could not delete by query for Solr collection [" + collectionString + "]", e);
+        }
+    }
+
+    private SolrClient uniProtSolrClient() {
+        String zookeeperhost = config.getZkHost();
+        if (zookeeperhost != null && !zookeeperhost.isEmpty()) {
+            String[] zookeeperHosts = zookeeperhost.split(",");
+
+            CloudSolrClient client =
+                    new CloudSolrClient.Builder(asList(zookeeperHosts), Optional.empty())
+                            .withConnectionTimeout(config.getConnectionTimeout())
+                            .withSocketTimeout(config.getSocketTimeout())
+                            .build();
+
+            client.setIdField(
+                    "accession_id"); // TODO: 10/07/19 refactor schemas to include unique field 'id'
+            return client;
+        } else if (!config.getHttphost().isEmpty()) {
+            return new HttpSolrClient.Builder()
```

Comment:
This line causes no errors, but internally it helps Solr when streaming results.

it would be better if each collection called `setIdField` with their specific ID value. So I'll create a Jira for this -- to refactor the code to ensure the idField is set (alternatively, we can keep the code teh same and refactor the schemas to all have the same unique ID field).